src/components/app/details/cdDetails/CDDetails.tsx

```diff
@@ -138,11 +138,16 @@ export default function CDDetails() {
             value: `${item.environmentId}`,
             label: item.environmentName,
             pipelineId: cdPipelinesMap.get(item.environmentId).id,
+            deploymentAppDeleteRequest: item.deploymentAppDeleteRequest,
         }
     })
 
-    if(result[0]['value'].result.length === 1 && !envId){
-      replace(generatePath(path, { appId, envId: envOptions[0].value, pipelineId: envOptions[0].pipelineId }))
+    const isEnvDeleted = result[0]['value'].result.find(
+        (_res) => _res.deploymentAppDeleteRequest,
+    ).deploymentAppDeleteRequest
+
+    if (result[0]['value'].result.length === 1 && !envId && !isEnvDeleted) {
```

Comment:
what if results is empty? what if result doesnt have value ?

Reply:
or does contract ensure it will not happen?

Reply:
Added safe check